utils/python/CIME/hist_utils.py

```diff
@@ -0,0 +1,172 @@
+"""
+Functions for actions pertaining to history files.
+"""
+
+from CIME.XML.standard_module_setup import *
+from CIME.XML.component import Component
+
+import logging, glob, os, shutil, re
+
+def _iter_model_file_substrs():
+    drv_comp = Component()
+    models = drv_comp.get_valid_model_components()
+
+    for model in models:
+        model = "cpl" if model == "DRV" else model.lower()
+        yield model
+
+def _get_all_hist_files(testcase, model, suffix="", from_dir=os.getcwd()):
+    suffix = (".%s" % suffix) if suffix else ""
+    test_hists = glob.glob("%s/%s.%s*.h?.nc%s" % (from_dir, testcase, model, suffix))
```

Comment:
@billsacks , that's a good point. Let's not forgot that we are running on modern machines and we should be able to do at least 4-8 cprnc's in parallel.